apm-sniffer/optional-producer-plugins/kafka-producer-plugin/src/main/java/org/apache/skywalking/apm/agent/core/kafka/KafkaTraceSegmentService.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.apm.agent.core.kafka;
+
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.skywalking.apm.agent.core.boot.BootService;
+import org.apache.skywalking.apm.agent.core.boot.OverrideImplementor;
+import org.apache.skywalking.apm.agent.core.boot.ServiceManager;
+import org.apache.skywalking.apm.agent.core.conf.Config;
+import org.apache.skywalking.apm.agent.core.context.TracingContext;
+import org.apache.skywalking.apm.agent.core.context.TracingContextListener;
+import org.apache.skywalking.apm.agent.core.context.trace.TraceSegment;
+import org.apache.skywalking.apm.agent.core.logging.api.ILog;
+import org.apache.skywalking.apm.agent.core.logging.api.LogManager;
+import org.apache.skywalking.apm.agent.core.remote.TraceSegmentServiceClient;
+import org.apache.skywalking.apm.network.language.agent.v3.SegmentObject;
+
+@OverrideImplementor(TraceSegmentServiceClient.class)
+public class KafkaTraceSegmentService implements BootService, TracingContextListener {
+    private static final ILog logger = LogManager.getLogger(KafkaTraceSegmentService.class);
+
+    private String topic;
+    private KafkaProducer<String, Bytes> producer;
+
+    @Override
+    public void prepare() throws Throwable {
+        topic = Config.Collector.Kafka.TOPIC_SEGMENT;
+    }
+
+    @Override
+    public void boot() throws Throwable {
+        producer = ServiceManager.INSTANCE.findService(MessageQueueServiceManagement.class).getProducer(topic);
+    }
+
+    @Override
+    public void onComplete() throws Throwable {
+        TracingContext.ListenerManager.add(this);
+    }
+
+    @Override
+    public void shutdown() throws Throwable {
+        TracingContext.ListenerManager.remove(this);
+    }
+
+    @Override
```

Comment:
need try catch exception and send trace data by async way? put to buffer, and consume to send by kafka, just advice.

Reply:
Thanks for your suggestions. 
Yes, we have to catch the exception.

Reply:
@dmsolr  any update?

Reply:
Yes, it will be released in 8.1.0 :)